--- lib/src/phy/modem/demod_soft.c
+++ lib/src/phy/modem/demod_soft.c
@@ -745,8 +745,8 @@ void demod_64qam_lte_s(const cf_t* symbols, short* llr, int nsymbols)

     llr[6 * i + 0] = -yre;
     llr[6 * i + 1] = -yim;
-    llr[6 * i + 2] = abs(yre) - 4 * SCALE_SHORT_CONV_QAM64 / sqrtf(42);
-    llr[6 * i + 3] = abs(yim) - 4 * SCALE_SHORT_CONV_QAM64 / sqrtf(42);
+    llr[6 * i + 2] = fabsf(yre) - 4 * SCALE_SHORT_CONV_QAM64 / sqrtf(42);
+    llr[6 * i + 3] = fabsf(yim) - 4 * SCALE_SHORT_CONV_QAM64 / sqrtf(42);
     llr[6 * i + 4] = abs(llr[6 * i + 2]) - 2 * SCALE_SHORT_CONV_QAM64 / sqrtf(42);
     llr[6 * i + 5] = abs(llr[6 * i + 3]) - 2 * SCALE_SHORT_CONV_QAM64 / sqrtf(42);
   }
@@ -768,8 +768,8 @@ void demod_64qam_lte_b(const cf_t* symbols, int8_t* llr, int nsymbols)

     llr[6 * i + 0] = -yre;
     llr[6 * i + 1] = -yim;
-    llr[6 * i + 2] = abs(yre) - 4 * SCALE_BYTE_CONV_QAM64 / sqrtf(42);
-    llr[6 * i + 3] = abs(yim) - 4 * SCALE_BYTE_CONV_QAM64 / sqrtf(42);
+    llr[6 * i + 2] = fabsf(yre) - 4 * SCALE_BYTE_CONV_QAM64 / sqrtf(42);
+    llr[6 * i + 3] = fabsf(yim) - 4 * SCALE_BYTE_CONV_QAM64 / sqrtf(42);
     llr[6 * i + 4] = abs(llr[6 * i + 2]) - 2 * SCALE_BYTE_CONV_QAM64 / sqrtf(42);
     llr[6 * i + 5] = abs(llr[6 * i + 3]) - 2 * SCALE_BYTE_CONV_QAM64 / sqrtf(42);
   }
